[HIGH PRIORITY] Missing log points on REST API failures, 404, 500, 400

The server should generate an error log if it's returning a 500 internal error, or a 400 Bad Request

get transcript and incoming endpoints


500 error triggering an event.

also see 500


[07/01/2021 01:12:06 PM] [ERROR] in utils.rocketchat: [Session_ID: <no session id set>] Failed to trigger event 500 Server Error: Internal Server Error for url: https://chatbot.vega.viasat.com/api/apps/public/21b7d3ba-031b-41d9-8ff2-fbbfa081ae90/incoming
--
 

<br class="Apple-interchange-newline">


[07/01/2021 01:12:06 PM] [ERROR] in utils.rocketchat: [Session_ID: <no session id set>] Failed to trigger event 500 Server Error: Internal Server Error for url: https://chatbot.vega.viasat.com/api/apps/public/21b7d3ba-031b-41d9-8ff2-fbbfa081ae90/incoming
--
 

<br class="Apple-interchange-newline">

--- a/app/apps/server/bridges/api.js
+++ b/app/apps/server/bridges/api.js
@@ -105,11 +105,11 @@ export class AppApisBridge {
 					res.status(status);
 					res.send(content);
 					if (status !== 200) {
-						this.orch.debugLog(`The endpoint /${ endpoint.path } of the App ${ appId } failed. \n${ JSON.stringify(request) }\nStatus Code: ${ status } Response: ${ JSON.stringify(content) }`);
+						this.orch.errorLog(`The endpoint /${ endpoint.path } of the App ${ appId } failed. \n${ JSON.stringify(request) }\nStatus Code: ${ status } Response: ${ JSON.stringify(content) }`);
 					}
 				})
 				.catch((reason) => {
-					this.orch.debugLog(reason);
+					this.orch.errorLog(reason);
 					// Should we handle this as an error?
 					res.status(500).send(reason.message);
 				});

--- a/app/apps/server/bridges/api.js
+++ b/app/apps/server/bridges/api.js
@@ -104,8 +104,12 @@ export class AppApisBridge {
 					res.set(headers);
 					res.status(status);
 					res.send(content);
+					if (status !== 200) {
+						this.orch.debugLog(`The endpoint /${ endpoint.path } of the App ${ appId } failed. \n${ JSON.stringify(request) }\nStatus Code: ${ status } Response: ${ JSON.stringify(content) }`);
+					}
 				})
 				.catch((reason) => {
+					this.orch.debugLog(reason);
 					// Should we handle this as an error?
 					res.status(500).send(reason.message);
 				});

--- a/app/apps/server/orchestrator.js
+++ b/app/apps/server/orchestrator.js
@@ -119,6 +119,12 @@ export class AppServerOrchestrator {
 		}
 	}
 
+	errorLog(...args) {
+		if (this.isDebugging()) {
+			this.getRocketChatLogger().error(...args);
+		}
+	}
+
 	getMarketplaceUrl() {
 		return this._marketplaceUrl;
 	}

